src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+/**
+ * Deadline class represents the task with a deadline.
+ *
+ * @author Rosa Kang
+ */
+public class Deadline extends Task {
+    /**
+     * Store deadline of the task.
+     */
+    protected String by;
+
+    /**
+     * Constructor for Deadline.
+     *
+     * @param description a description of the deadline task
+     * @param by          a deadline date
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Constructor for Deadline.
+     *
+     * @param description description of the deadline task
+     * @param by          deadline date
+     * @param isDone      completion of the task
+     */
+    public Deadline(String description, String by, boolean isDone) {
+        super(description, isDone);
+        this.by = by;
+    }
+
+    /**
+     * format Deadline to be stored into txt file.
+     *
+     * @return String representation of Deadline for txt file
```

Comment:
Good summary of the return statement. Makes it easier to understand the use and reason for the String being returned from this method.